### PR TITLE
tests: Update tests to newer constrained routes

### PR DESCRIPTION
**Pull request by brainrepo**
Update tests to the newer constrained routes in order to remove warning. #2889 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by climba03003 — COMMENTED** [review]
It does remove the warning, but warning message also need to be tested.

File: test/versioned-routes.test.js [hunk 1]
```diff
@@ -569,3 +577,38 @@ test('Vary header check (for documentation example)', t => {
     t.strictEqual(res.headers.vary, undefined)
   })
 })
+
+test('Should trigger a warning when a versioned route is registered via version option', t => {
```
Comment (jsumners) [inline test/versioned-routes.test.js:581]: Should there be a `t.tearDown` for this test to remove the proxying?

Comment (mcollina) [reply test/versioned-routes.test.js:581]: There is no need, it does not change the global cache.

**Review by mcollina — APPROVED** [review]
lgtm

File: test/versioned-routes.test.js (hunk 1 above)
Comment (jsumners) [reply test/versioned-routes.test.js:581]: 👍 

I'm accustomed to `mock-require`.

**Review by jsumners — APPROVED** [review]
LGTM

**Comment by airhorns** [comment]
Apologies for introducing these ,thanks @brainrepo for the fix!

**Pull request merged**